src/HomePage.tsx

```diff
@@ -1,29 +1,32 @@
 import { render } from '@testing-library/react';
 import { resolve } from 'dns';
 import React from 'react';
+import { connect } from 'react-redux';
 import Button from '@mui/material/Button';
 import Avatar from '@mui/material/Avatar';
 import { NavLink } from 'react-router-dom';
 import { getCharacterByName, getCharacters } from './api';
 import Preloader from './Preloader';
 import styles from './homePage.module.css';
 import logo from './common/Marvel_Studios_logo.png';
+import { loadingHeroesActionCreatorByName } from './redux/heroesReduser';
+import { HeroesInitialState, loadingHeroesActionCreator } from './redux/heroesReduser';
 
-interface HomeState {
+export interface HomeState {
   heroes: Array<Heroes>,
   loading: boolean,
   inputValue: string,
   location: any,
-  history: any
+  history: any,
+  loadingHeroesDispatchDefault: any,
+  loadingHeroesDispatchByName: any
```

Comment:
no sense to make 2 separate functions for this, just pass query name as a param to the func